src/main/kotlin/no/nav/tilleggsstonader/sak/vedtak/barnetilsyn/TilsynBarnVedtakRepository.kt

```diff
@@ -1,24 +1,50 @@
 package no.nav.tilleggsstonader.sak.vedtak.barnetilsyn
 
+import no.nav.tilleggsstonader.sak.infrastruktur.database.Sporbar
+import no.nav.tilleggsstonader.sak.infrastruktur.database.repository.InsertUpdateRepository
+import no.nav.tilleggsstonader.sak.infrastruktur.database.repository.RepositoryInterface
+import no.nav.tilleggsstonader.sak.vedtak.TypeVedtak
+import no.nav.tilleggsstonader.sak.vedtak.VedtakRepository
 import org.springframework.data.annotation.Id
+import org.springframework.data.relational.core.mapping.Embedded
 import org.springframework.stereotype.Repository
 import java.util.UUID
 
 @Repository
-class TilsynBarnVedtakRepository {
-    /**
-     * TODO Erstatt disse metodene når det er et faktiskt repository
-     */
-    fun findByIdOrNull(behandlingId: UUID): VedtakTilsynBarn? = null
-
-    fun insert(vedtak: VedtakTilsynBarn): VedtakTilsynBarn = vedtak
-
-    fun deleteById(behandlingId: UUID) {}
-}
+interface TilsynBarnVedtakRepository :
+    VedtakRepository<VedtakTilsynBarn>,
+    RepositoryInterface<VedtakTilsynBarn, UUID>,
+    InsertUpdateRepository<VedtakTilsynBarn>
 
+/**
+ * Trenger vi noe mer enn data her? Kan den kanskje dekke alle tilfeller?
+ * Eller om ma har vedtak, og beregningsgrunnlag som et eget?
+ * Trenger man begrunnelse som eget felt?
+ */
 data class VedtakTilsynBarn(
```

Comment:
Denne burde kanskje flyttes til egen fil? Ser for meg det er en klasse jeg vil se på relativt ofte, og gjene vil kunne søke opp på filnavn 🤠

Reply:
Ja, tenkte repositoriet var så litet og de er veldig tett knyttet 😬 flytter, så kan vi eventuellt flytte mappers til den fila og
Anbefaler å søke etter klassenavn 👀